vital/types/track.h

```diff
@@ -242,6 +250,12 @@ class VITAL_EXPORT track : public std::enable_shared_from_this<track>
   ///         otherwise \c false.
   bool contains( frame_id_t frame ) const;
 
+  /// Erase track state at iterator.
+  ///
+  /// \return An iterator to the item which followed the erased item.
+  history_const_itr erase( history_const_itr iter )
```

Comment:
Actually, my other comment about resetting the backward pointer from the track state to the track probably applies better here.